tests/phpunit/tests/test-factory.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+class Test_WPNotify_Factory extends WPNotify_TestCase {
+
+
+	public function test_create_with_vendor_sender() {
+
+		$vendor_sender = $this->createMock( 'WPNotify_Sender' );
+
+		$message_factory = $this->getMockBuilder( 'WPNotify_MessageFactory' )
+		                        ->setMethods( array( 'create', 'accepts' ) )
+		                        ->getMock();
+
+		$message_factory->method( 'create' )->willReturn( $this->createMock( 'WPNotify_Message' ) );
+		$message_factory->method( 'accepts' )->willReturn( true );
+
+		$sender_factory = $this->getMockBuilder( 'WPNotify_SenderFactory' )
+		                       ->setMethods( array( 'create' ) )
+		                       ->getMock();
+
+		$sender_factory->method( 'create' )->willReturn( $this->createMock( 'WPNotify_Sender' ) );
```

Comment:
Good. With all these mocks, you won't have to rewrite your test each time we change the code of any of these factories.